spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/VaultEnvironmentRepository.java

```diff
@@ -291,4 +295,45 @@ public void setRenewable(boolean renewable) {
 			this.renewable = renewable;
 		}
 	}
+
+	static class VersionedVaultResponse extends VaultResponse {
```

Comment:
How about combining URL and response processing in a strategy object? We could have `VaultKvAccessStrategy` that accepts `RestOperations` and a path and returns a `VaultResponse` object. Using strategy objects is a bit more focused by keeping all functionality in a single place for a particular strategy without the need to distribute behavior across multiple types (`VersionedVaultResponse`, `VaultVersioning`, making `data` `protected`).